demo/unload_event_test/run.py

```diff
@@ -0,0 +1,34 @@
+"""This demo is only meant to test the unload event.
+It will write to a file when the unload event is triggered.
+May not work as expected if multiple people are using it.
+"""
+import gradio as gr
+from pathlib import Path
+
+log_file = (Path(__file__).parent / "output_log.txt").resolve()
+
+
+def test_fn(x):
+    with open(log_file, "a") as f:
+        f.write(f"incremented {x}\n")
+    return x + 1, x + 1
+
+def delete_fn(v):
+    with log_file.open("a") as f:
+        f.write(f"deleted {v}\n")
+
+def unload_fn(req: gr.Request):
+   assert req.request
```

Comment:
What's the purpose of the `gr.Request` here? For simplicity, maybe better to remove it
```suggestion
def unload_fn():
```

Reply:
Just to show that you can pass gr.Request to state but can remove